admin/src/containers/Login/Login.js

```diff
@@ -6,35 +6,39 @@ import Login from '../../components/Login/Login';
 import session from '../../helpers/Session';
 import config from '../../config';
 
-const LoginContainer = withRouter(props => {
+export const LoginContainer = props => {
   const { from } = props.location.state || { from: { pathname: '/' } };
   const { history } = props;
 
   const handleLogin = (username, password) => {
-    fetch(`${config.API_URL}/api/management/users/login`, {
-      method: 'POST',
-      body: JSON.stringify({
-        username,
-        password
-      }),
-      headers: {
-        'Content-Type': 'application/json'
-      }
-    })
-      .then(async res => {
-        session.set(await res.json());
-        history.push(from);
+    return new Promise((resolve, reject) => {
```

Comment:
Fetch already returns a promise so I suggest we return it instead of creating a new one. We risk getting into the [forgotten promise antipattern](http://taoofcode.net/promise-anti-patterns/) – not that the code will not work.

Reply:
yeah there is no reason to create a new promise here, the fetch one is better 👍